src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java

```diff
@@ -92,14 +92,24 @@ public void addDetailLine(ConcurStandardAccountingExtractDetailLine detailLine)
 
     protected void addCorpCardPersonalExpenseDetailLine(ConcurStandardAccountingExtractDetailLine detailLine) {
         String accountingFieldsKey = buildAccountingFieldsKeyForCorpCardPersonalExpense(detailLine);
-        Map<String, List<ConcurStandardAccountingExtractDetailLine>> consolidatedLinesMap;
+        Map<String, List<ConcurStandardAccountingExtractDetailLine>> consolidatedLinesMap = getConsolidatedPersonalExpensesMapForLine(detailLine);
+        consolidatedLinesMap.computeIfAbsent(accountingFieldsKey, (key) -> new ArrayList<>())
+                .add(detailLine);
+    }
+
+    protected Map<String, List<ConcurStandardAccountingExtractDetailLine>> getConsolidatedPersonalExpensesMapForLine(
+            ConcurStandardAccountingExtractDetailLine detailLine) {
         if (detailLine.getJournalAmount().isPositive()) {
-            consolidatedLinesMap = consolidatedCorpCardPersonalExpenseDebits;
+            if (collectorHelper.lineRepresentsReturnOfCorporateCardPersonalExpenseToUser(detailLine)) {

```

Comment:
I am thinking reorganizing this a little might make it easier to read.  The if/else levels make it confusing to read and it is starting to get long.  Maybe something like
if (detailLine.getJournalAmount().isPositive()) {
  getConsolidatedPersonalExpensesMapForPositiveLine(line);
} else {
  getConsolidatedPersonalExpensesMapForNegativeLine(line)
}

and move the contents if the if else into the new functions?